src/main/java/it/pagopa/transactions/commands/handlers/TransactionActivateHandler.java

```diff
@@ -238,39 +244,47 @@ public Mono<Tuple2<Mono<TransactionActivatedEvent>, String>> handle(
-                ZonedDateTime paymentTokenValidityEnd = transactionActivation
-                        .plus(Duration.ofSeconds(paymentTokenTimeout));
-                Duration paymentTokenValidityTimeLeft = Duration.between(ZonedDateTime.now(), paymentTokenValidityEnd);
-                span.startSpan("transaction", "activation", "activation cached")
-                        .setName("Transaction re-activated")
-                        .setLabel("paymentToken", paymentRequestInfo.paymentToken())
-                        .setLabel("paymentTokenLeftTimeSec", paymentTokenValidityTimeLeft.getSeconds());
+        String paymentToken = paymentRequestInfo.paymentToken();
+        if (transactionActivationDateString != null && paymentToken != null) {
+            ZonedDateTime transactionActivation = ZonedDateTime.parse(transactionActivationDateString);
+            ZonedDateTime paymentTokenValidityEnd = transactionActivation
+                    .plus(Duration.ofSeconds(paymentTokenTimeout));
+            Duration paymentTokenValidityTimeLeft = Duration.between(ZonedDateTime.now(), paymentTokenValidityEnd);
 
-                log.info(
-                        "PaymentRequestInfo cache hit for {} with valid paymentToken {}. Validity left time: {}",
-                        paymentRequestInfo.id(),
-                        paymentRequestInfo.paymentToken(),
-                        paymentTokenValidityTimeLeft
-                );
-            } else {
-                log.error(
-                        "Cannot trace repeated transaction activation for {} with payment token: {}, missing transaction activation date",
-                        paymentRequestInfo.id(),
-                        paymentRequestInfo.paymentToken()
-                );
-                span.captureException(
-                        new IllegalArgumentException(
-                                "Null transaction activation date for rptId %s in repeated activation"
-                                        .formatted(paymentRequestInfo.id().toString())
-                        )
-                );
-            }
-        } finally {
-            span.end();
+            /*
+             * Issue https://github.com/elastic/kibana/issues/123256 Span events attached to
+             * the Span.currentSpan() are not visible into Transaction detail so here we
+             * start a new span as workaround in order to make this event visible also
+             * inside Transaction view
+             */
+            openTelemetryUtils.addSpanWithAttributes(
+                    "Transaction re-activated",
+                    Attributes.of(
+                            AttributeKey.stringKey("paymentToken"),
+                            paymentToken,
+                            AttributeKey.longKey("paymentTokenLeftTimeSec"),
+                            paymentTokenValidityTimeLeft.getSeconds()
```

Comment:
The attribute keys should be constant declared somewhere

Reply:
Fixed with https://github.com/pagopa/pagopa-ecommerce-transactions-service/pull/285/commits/025ff0ba1a017229b15020700c01b11865c310a6